// Copyright 2021 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package google.cloud.cpp.generator;

message ServiceConfiguration {
  string service_proto_path = 1;
  string product_path = 2;
  string initial_copyright_year = 3;
  repeated string omitted_rpcs = 4;
  string service_endpoint_env_var = 5;
  string emulator_endpoint_env_var = 6;
  // Generating Async APIs is typically unnecessary for RPCs that are
  // non-streaming, non-paginated, and not longrunning operations. If we need
  // to generate an Async API for such an RPC, we can list it here.
  //
  // Only the Async API will be generated for RPCs listed both here and in
  // `omitted_rpcs`.
  repeated string gen_async_rpcs = 7;
}

message GeneratorConfiguration {
  repeated ServiceConfiguration service = 1;
}
